pymc_extras/statespace/models/structural/core.py

```diff
@@ -519,54 +542,116 @@ def __init__(
         share_states: bool = False,
     ):
         self.name = name
-        self.k_endog = k_endog
-        self.k_states = k_states
         self.share_states = share_states
-        self.k_posdef = k_posdef
+        self._k_posdef = k_posdef
         self.measurement_error = measurement_error
 
-        self.state_names = list(state_names) if state_names is not None else []
-        self.observed_state_names = (
-            list(observed_state_names) if observed_state_names is not None else []
+        self.param_info = ParameterInfo(
+            parameters=[
+                Parameter(name=n, shape=(1,), dims=(f"{n}_placeholder"))
+                for n in (param_names or [])
```

Comment:
```suggestion
                for name in param_names
```

Have the class properties always return a list, so you don't have to check for `None` here.